### PR TITLE
Add the capability to update discovery configuration of the primary organization.

### PR DESCRIPTION
**Pull request by Yasasr1**
## Purpose
This pr introduces a new method for updating the discovery configuration in the organization configuration management service. 

## Related issue
- https://github.com/wso2/product-is/issues/21572

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.config.service/src/main/java/org/wso2/carbon/identity/organization/config/service/constant/OrganizationConfigConstants.java [hunk 1]
```diff
@@ -18,7 +18,6 @@
 
 package org.wso2.carbon.identity.organization.config.service.constant;
```
Comment (dhaura) [inline components/org.wso2.carbon.identity.organization.config.service/src/main/java/org/wso2/carbon/identity/organization/config/service/constant/OrganizationConfigConstants.java:19]: Let's update the license header. Check other places as well.

File: components/org.wso2.carbon.identity.organization.config.service/src/main/java/org/wso2/carbon/identity/organization/config/service/OrganizationConfigManagerImpl.java [hunk 2]
```diff
@@ -75,6 +78,26 @@ public void addDiscoveryConfiguration(DiscoveryConfig discoveryConfig) throws Or
         }
     }
 
+    @Override
+    public void updateDiscoveryConfiguration(DiscoveryConfig discoveryConfig) throws OrganizationConfigException {
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.config.service/src/main/java/org/wso2/carbon/identity/organization/config/service/OrganizationConfigManagerImpl.java:82]: Add a unit test

File: components/org.wso2.carbon.identity.organization.config.service/src/main/java/org/wso2/carbon/identity/organization/config/service/OrganizationConfigManagerImpl.java [hunk 3]
```diff
@@ -75,6 +78,26 @@ public void addDiscoveryConfiguration(DiscoveryConfig discoveryConfig) throws Or
...
+                throw handleClientException(ERROR_CODE_DISCOVERY_CONFIG_UPDATE_NOT_ALLOWED);
+            }
+            Optional<Resource> resourceOptional = getDiscoveryResource();
+            Resource resource = buildResourceFromValidationConfig(discoveryConfig);
+            if (!resourceOptional.isPresent()) {
+                getConfigurationManager().addResource(RESOURCE_TYPE_NAME, resource);
+            } else {
+                getConfigurationManager().replaceResource(RESOURCE_TYPE_NAME, resource);
+            }
+
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.config.service/src/main/java/org/wso2/carbon/identity/organization/config/service/OrganizationConfigManagerImpl.java:95]:
```suggestion
```

File: components/org.wso2.carbon.identity.organization.config.service/src/test/java/org/wso2/carbon/identity/organization/config/service/OrganizationConfigManagerImplTest.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+public class OrganizationConfigManagerImplTest {
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.config.service/src/test/java/org/wso2/carbon/identity/organization/config/service/OrganizationConfigManagerImplTest.java:61]: add class comment

File: components/org.wso2.carbon.identity.organization.config.service/src/test/java/org/wso2/carbon/identity/organization/config/service/util/TestUtils.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import org.apache.commons.dbcp.BasicDataSource;
+import org.apache.commons.lang.StringUtils;
+
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class TestUtils {
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.organization.config.service/src/test/java/org/wso2/carbon/identity/organization/config/service/util/TestUtils.java:30]: add class comment

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**